src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -346,15 +346,9 @@ void snc_quota_manager::get_or_create_quota_context(
     }
 }
 
-snc_quota_manager::delays_t snc_quota_manager::get_shard_delays(
-  snc_quota_context& ctx, const clock::time_point now) const {
+snc_quota_manager::delays_t snc_quota_manager::get_shard_delays() const {
```

Comment:
It looks like this should check `ctx->_exempt`, am I missing something?

Reply:
Yeah, I agree it looks like we should skip throttling on `ctx->_exempt`. The original [PR](https://github.com/redpanda-data/redpanda/pull/10755) working even says:

> Connections presenting a matching client_id value will not have the size of their requests accounted for throughput control, **not they will be throttled.**

I've pushed the fix here in this PR now: https://github.com/redpanda-data/redpanda/pull/18218/commits/ebb9af387c68b1c364b3318fd09f3419d0edf461. It's a small and related fix so I've left it in this PR, but I'm open to extracting it into a separate PR if you think it's worth it.